parachain/pallets/ethereum-beacon-client/src/tests_minimal.rs

```diff
@@ -217,6 +219,52 @@ mod beacon_minimal_tests {
 		});
 	}
 
+	#[test]
+	fn it_processes_pruning_finalized_header_update() {
+		let initial_sync = get_initial_sync::<mock_minimal::Test>();
```

Comment:
I am not sure if this is possible but maybe we should drop the bounds configured `mock_minimal::Test` to small amounts. And then just import some test data to reproduce states.

For instance, set `MaxExecutionHeaderCachedSize` to 2 and then try to import 3 headers. The same for `MaxSyncCommitteeCachedSize`. Unless it is too hard to get into those states.

Reply:
Drop all bounds to 1 and all tests still passed as expected. Consider we'll import `initialSync`  so we have 2 headers to cover the case you mentioned above. 

Btw: actually in this test case we'll fill `FinalizedBeaconHeadersList` to `MaxBounds` first before `import_finalized_header` as following, so will also cover the case even with a big value.

https://github.com/Snowfork/snowbridge/blob/92e5f0f7478fe6c473d69a1ed329fe026df80ca2/parachain/pallets/ethereum-beacon-client/src/tests_minimal.rs#L242-L254

Reply:
Drop all bounds and more tests: https://github.com/Snowfork/snowbridge/pull/794/commits/f6de4ca9df00c18f80464166749cfe32486a2797